Source/WebKit/GPUProcess/media/RemoteVideoFrameObjectHeap.cpp

```diff
@@ -91,13 +91,31 @@ void RemoteVideoFrameObjectHeap::close()
     // TODO: add can happen after stopping.
 }
 
-RemoteVideoFrameProxy::Properties RemoteVideoFrameObjectHeap::add(Ref<WebCore::VideoFrame>&& frame)
+RemoteVideoFrameObjectHeap::AddWithCompletionResult RemoteVideoFrameObjectHeap::addWithCompletion(Ref<WebCore::VideoFrame>&& videoFrame)
 {
-    auto newFrameReference = RemoteVideoFrameReference::generateForAdd();
-    auto properties = RemoteVideoFrameProxy::properties(newFrameReference, frame);
-    auto success = m_heap.add(newFrameReference, WTFMove(frame));
-    ASSERT_UNUSED(success, success);
-    return properties;
+    // High bit of the identifiers are reserved for identifiers allocated in the object holder namespace, e.g. GPU process.
+    static std::atomic<uint64_t> rawIdentifier;
+    RemoteVideoFrameIdentifier identifier { ++rawIdentifier | (static_cast<uint64_t>(1) << 63) };
```

Comment:
Maybe we should add a GPUProcess generate method  and a WebProcess generate method to make the code code clearer.